design/src/rfcs/rfc0017_logging_sensitive.md

```diff
@@ -0,0 +1,376 @@
+
+| Trait                                                                                                        | Configurable        |
+|--------------------------------------------------------------------------------------------------------------|---------------------|
+| [http](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#http-trait)                           | URI and Status Code |
+| [httpHeader](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httpheader-trait)               | Headers             |
+| [httpPrefixHeaders](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httpprefixheaders-trait) | Headers             |
+| [httpLabel](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httplabel-trait)                 | URI                 |
+| [httpQuery](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httpquery-trait)                 | Query Parameters    |
+| [httpResponseCode](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httpresponsecode-trait)   | Status Code
+
+Each of these configurable parts must therefore be logged cautiously.
+
+### Scope and Guidelines
+
+Making the logging of sensitive data forbidden an type theoretic invariant is unfeasible. With the current API, the customer will always have an opportunity to log a request containing sensitive data before it enters the `Service<Request<B>>` that we provide to them.
+
+```rust
+// The API provides us with a `Service<Request<B>>`
+let app: Router = OperationRegistryBuilder::default().build().expect("unable to build operation registry").into();
+
+// We can `ServiceExt::map_request` log a request with potentially sensitive data
+let app = app.map_request(|request| {
+        info!(?request);
+        request
+    });
+```
+
+A more subtle violation of the specification may occur when the customer enables verbose logging - a third-party dependency might simply log data marked as sensitive, for example `tokio` or `hyper`.
+
+These two cases illustrate that `smithy-rs` can only prevent violation of the specification in a restricted scope - logs emitted from generated code and the runtime crates. A `smithy-rs` specific guideline should be available to the customer which outlines how to avoid violating the specification in areas outside of our control.
+
+### Routing
+
+The sensitivity and HTTP bindings are declared within specific structures/operations. For this reason, in the general case, it's unknowable whether or not any given part of a request is sensitive until we determine which operation is tasked with handling the request and hence which fields are bound. Implementation wise, this means that any `Layer` applied _before_ routing has taken place cannot log anything sensitive without performing routing logic itself.
```

Comment:
> in the general case

in noted because in the case where there are absolutely no `@sensitive` traits used in the model we can log freely anywhere.